pkg/network/encoding/http2.go

```diff
@@ -0,0 +1,175 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package encoding
+
+import (
+	"github.com/gogo/protobuf/proto"
+
+	model "github.com/DataDog/agent-payload/v5/process"
+
+	"github.com/DataDog/datadog-agent/pkg/network"
+	"github.com/DataDog/datadog-agent/pkg/network/protocols/http"
+)
+
+type http2Encoder struct {
```

Comment:
This is the same as `httpEncoder` currently, so can we just use `httpEncoder` instead? When assigning to the `http2Aggregrations` field in `encoding.go`, we could just create a `model.HTTP2Aggregation` object with the `EndpointAggregations` member from `model.HTTPAggregation`.

Reply:
This change requires a significant refactoring effort, as illustrated in the following image:

<img width="1838" alt="Screenshot 2023-02-28 at 10 26 18" src="https://user-images.githubusercontent.com/108348428/221797220-7c2aa01e-6f60-449d-bb80-3913033b091c.png">

The image demonstrates that we are currently using different payloads between the encoders. It is not just a matter of different protos, but rather different payloads.
We acknowledge that we need to create a refactoring plan for our code to handle the use of different protocols in USM.

Reply:
indeed as Amit said, we are currently facing that we can do much better in the user mode for everything related to ease the support of new protocols for decoding, we plan to do so after adding http2 and kafka